### DISCUSSION TITLE
Any way to specify port visibility in devcontainer.json?

### DISCUSSION THREAD

**Discussion by deepansh96** [community/community#10394] [Codespaces] [answered]
Currently, I can specify which ports I want to auto forward using the `devcontainer.json` file and I can also label specific ports.
[Official VSCode Reference](https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_general-devcontainerjson-properties) 

But I didn't see any way to specify visibility of ports in the docs. I have to manually go and change the visibility of each port from the GUI. 

Let me know if I'm looking in the wrong place and the support for this exists.

Thanks!

**Comment by asciimike** [comment, marked as answer]
Not currently, though this is something we should definitely consider!

My first thought would be to add a `visibility: public | org | private` option to `portsAttributes` property:

```json
{
    ...
    "forwardPorts": [3000],
    "portsAttributes": {
        "3000": {
	    "visibility": "public"
	}
    }
    ...
}
```

I've opened an issue internally to discuss the above.

**Reply by r3dpointsp** [reply]
Really excited for this feature since without it our ability to use CodeSpaces is limited. We have SaaS webhooks that need to interact with our app in a CodeSpace and controlling public port visibility is needed for that use case.

**Reply by mariachiinajar** [reply]
Hi, 
Just found this thread more than a year later since your comment. 
Has this been implemented in the current `Codespace` version? 
I can't seem to find any information elsewhere about configuring port visibility via `devcontainer` other than this Issue. Thank you!

**Reply by treeder** [reply]
No, you still cannot do this.

**Reply by 825i** [reply]
It's possible. My solution is below.

**Comment by treeder** [comment]
Dupe: https://github.com/github/feedback/discussions/4068

**Comment by 825i** [comment]
Hi, the documentation isn't the best with this sadly. Specify your ports as follows depending on your needs/uses:

devcontainer.json:
```
"appPort": [ 5000, "8921:5005" ]
```

docker-compose:
```
ports:
- "5000"
- "8921:5005"
```

Dockerfile:
```
"forwardPorts": [ 5000, "8921:5005" ]
```

This method allows you to share Codespaces and mappings, provided all team members are using the same _devcontainer.json_ and the ports are properly exposed in Docker.

The quick and dirty way is to have a copy in your repository with the _devcontainer.json_ and _Dockerfile_ which you expect your devs to use, which can be distributed via git.

**Reply by lbarros-arista** [reply]
Worked like a charm thanks!!!
